### PR TITLE
Conditional flows new tests

### PR DESCRIPTION
**Pull request by mastepan**
//test: `@integrations-conditional-flows`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:3:8: Unused import - com.codeborne.selenide.Condition. [UnusedImports]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:5: Wrong order for 'io.syndesis.qe.pages.SyndesisPageObject' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:7: 'lombok.extern.slf4j.Slf4j' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:8: Wrong order for 'org.openqa.selenium.By' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:12: Wrong order for 'com.codeborne.selenide.Condition.visible' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:14: Wrong order for 'org.assertj.core.api.Assertions.fail' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/other/
  ConditionalFlowSteps.java:10:8: Unused import - io.syndesis.qe.pages.integrations.editor.add.steps.ConditionalFlows. [UnusedImports]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:6: Wrong order for 'org.assertj.core.api.Assertions.fail' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:10: Wrong order for 'io.syndesis.qe.pages.SyndesisPageObject' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:15: Wrong order for 'org.openqa.selenium.By' import. [ImportOrder]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:6: Wrong order for 'org.assertj.core.api.Assertions.fail' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:10: Wrong order for 'io.syndesis.qe.pages.SyndesisPageObject' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/connection/
  ChooseAction.java:15: Wrong order for 'org.openqa.selenium.By' import. [ImportOrder]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Conditional flows - content base routing | Conditional flows - delete step 
Conditional flows - content base routing | Conditional flows - multiple flows 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: ui-tests/src/test/resources/features/integrations/conditional-flow.feature [hunk 1]
```diff
@@ -424,3 +520,36 @@ Feature: Conditional flows - content base routing
...
+      | usedefaultflow | false |
+    And click on the "Next" button
+    And force fill in values by element data-testid
+      | describe-data-shape-form-kind-input | JSON Instance |
+    And fill text into text-editor
+      | {"greeting":"Hi"} |
+    And click on the "Next" button
+
+    When configure condition on position 2
+    Then add a data mapping step - open datamapper
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/integrations/conditional-flow.feature]: Same as above

File: ui-tests/src/test/resources/features/integrations/conditional-flow.feature [hunk 2]
```diff
@@ -424,3 +520,36 @@ Feature: Conditional flows - content base routing
...
+    Then create data mapper mappings
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/integrations/conditional-flow.feature]: Same as above

File: ui-tests/src/test/resources/features/integrations/conditional-flow.feature [hunk 3]
```diff
@@ -424,3 +520,36 @@ Feature: Conditional flows - content base routing
...
+      | message | greeting |
+    And click on the "Done" button
+    Then check that there is no warning inside of step number "2"
+    Then return to primary flow from integration flow from dropdown
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/integrations/conditional-flow.feature]: @When

File: ui-tests/src/test/resources/features/integrations/conditional-flow.feature [hunk 4]
```diff
@@ -424,3 +520,36 @@ Feature: Conditional flows - content base routing
...
+
+    Then add a default flow through warning link - open flows configuration
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/integrations/conditional-flow.feature]: @When

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConditionalFlowSteps.java [hunk 5]
```diff
@@ -50,6 +55,22 @@ private static By getConditionIconsOnPosition(String position) {
...
+        $(EditFlowStepElements.WARNING_ICON).shouldBe(visible).click();
+        $(EditFlowStepElements.ADD_DATA_MAPPING_STEP_WARNING_LINK).shouldBe(visible).click();
+    }
+    @Then("^add a default flow through warning link - open flows configuration$")
+    public void addDefaultFlowThroughWarningLink() {
+        $(EditFlowStepElements.WARNING_ICON).shouldBe(visible).click();
+        $(EditFlowStepElements.ADD_DEFAULT_FLOW_WARNING_LINK).shouldBe(visible).click();
+    }
+
+    @Then("^Add another condition$")
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ConditionalFlowSteps.java]: also change to when here as @mkralik3 suggested

File: ui-tests/src/test/resources/features/integrations/conditional-flow.feature [hunk 6]
```diff
@@ -200,9 +204,96 @@ Feature: Conditional flows - content base routing
...
+    And click on the "Next" button
+    And click on the "Next" button
+
+    When configure condition on position 2
+    And add integration step on position "0"
+    And select the "PostgresDB" connection
+    And select "Invoke SQL" integration action
+    And fill in invoke query input with "insert into contact(first_name) values ('Shaco was here')" value
+    And click on the "Next" button
+    Then return to primary flow from integration flow directly
```
[the remark this answers is not in the document]
Comment (mastepan) [reply ui-tests/src/test/resources/features/integrations/conditional-flow.feature]: fixed

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**